x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportIsolateDatafeedAction.java

```diff
@@ -52,22 +52,26 @@ protected void doExecute(Task task, IsolateDatafeedAction.Request request, Actio
     protected IsolateDatafeedAction.Response newResponse(IsolateDatafeedAction.Request request, List<IsolateDatafeedAction.Response> tasks,
                                                          List<TaskOperationFailure> taskOperationFailures,
                                                          List<FailedNodeException> failedNodeExceptions) {
+        // We only let people isolate one datafeed at a time, so each list will be empty or contain one item
+        assert tasks.size() <= 1 : "more than 1 item in tasks: " + tasks.size();
+        assert taskOperationFailures.size() <= 1 : "more than 1 item in taskOperationFailures: " + taskOperationFailures.size();
+        assert failedNodeExceptions.size() <= 1 : "more than 1 item in failedNodeExceptions: " + failedNodeExceptions.size();
         if (taskOperationFailures.isEmpty() == false) {
             throw org.elasticsearch.ExceptionsHelper
                     .convertToElastic(taskOperationFailures.get(0).getCause());
         } else if (failedNodeExceptions.isEmpty() == false) {
             throw org.elasticsearch.ExceptionsHelper
                     .convertToElastic(failedNodeExceptions.get(0));
-        } else {
-            return new IsolateDatafeedAction.Response(false);
+        } else if (tasks.isEmpty() == false) {
+            return tasks.get(0);
         }
+        return new IsolateDatafeedAction.Response(false);
     }
 
     @Override
     protected void taskOperation(IsolateDatafeedAction.Request request, TransportStartDatafeedAction.DatafeedTask datafeedTask,
                                  ActionListener<IsolateDatafeedAction.Response> listener) {
         datafeedTask.isolate();
-        listener.onResponse(new IsolateDatafeedAction.Response(false));
+        listener.onResponse(new IsolateDatafeedAction.Response(true));
```

Comment:
seems weird to make this `true`, but the final response to be `false`. Seems like an arbitrary change.

Reply:
Previously the response was _always_ `false`, which was a bit crazy.

> but the final response to be `false`

The final response _shouldn't_ be false, because it should be returned by:

```
        } else if (tasks.isEmpty() == false) {
            return tasks.get(0);
        }
```

That last line of the `newResponse` method only runs if there were no responses and no errors, which seems impossible.  But I left it there just in case there is a way that can happen.  Maybe if the datafeed gets deleted in between validation of the isolate datafeed request and getting the response from it?

The alternative would be just to change the non-error response to always return `isolated: true` like acknowledged requests do.